### PR TITLE
correct the probability analysis description for the "pri < 1" case

### PR DESCRIPTION
**Pull request by yaojingguo**
@spencerkimball there is some problem with the probability analysis description for the "pri < 1" case in the comment.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3706)

### CODE REVIEW

File: roachpb/data.go [hunk 1]
```diff
@@ -661,12 +661,11 @@ func MakePriority(userPriority float64) int32 {
 	//
 	// For every random trial, if the normal priority is chosen >= max*x,
 	// the user priority loses 100% of the time; otherwise, wins 50% of
-	// the time. Therefore, x fraction of the time, loss is user(100%)
```
Comment (tschottdorf) [inline roachpb/data.go]: while you're here, rename this `x` to `y` since it's going to be different from the above `x` for `pri > 1`.

Comment (tschottdorf) [inline roachpb/data.go]: btw, line 656 could be written as ` = (pri-1)/(pri+1)`, which is easy to recognize again below.

File: roachpb/data.go [hunk 2]
```diff
@@ -661,12 +661,11 @@ func MakePriority(userPriority float64) int32 {
...
-	// and (1-x) fraction of the time win is user(50%).
+	// the time. Therefore, (1-x) fraction of the time, loss is user(100%)
+	// and x fraction of the time win is user(50%).
 	//
-	//   (1-x) * 1/2 = pri / (1 + pri)
-	//   -x/2 = pri / (1 + pri) - 1/2
-	//   x = 1 - 2*pri / (1 + pri)
+	//   x * 1/2 = pri / (1 + pri)
+	//   x = 2*pri / (1 + pri)
```
Comment (tschottdorf) [inline roachpb/data.go]: ... then here you can write `y = 2*pri / (1+pri) = (pri-1)/(pri+1) + 1`.

File: roachpb/data.go [hunk 3]
```diff
@@ -661,12 +661,11 @@ func MakePriority(userPriority float64) int32 {
...
 	x := 2*float64(userPriority)/(1+float64(userPriority)) - 1
```
Comment (tschottdorf) [inline roachpb/data.go]: `pri := float64(userPriority)`
`x := (pri - 1) / (pri + 1)`

**Comment by tschottdorf** [comment]
LGTM. You can ignore my suggestions above, I think they improve the presentation but if you disagree that's ok with me.

**Comment by yaojingguo** [comment]
@tschottdorf, a new commit after adopting your suggestions except the renaming from `x` to `y`. PTAL.

**Comment by tschottdorf** [comment]
LGTM

**Pull request merged**